doc/underpinnings.tex

```diff
@@ -0,0 +1,779 @@
+expectation percentage of leavers is stronger or weaker (more or less
+confident in) depending on the amount of data we have.  Intuitively if
+we have lots of observations around the number of leavers then we can
+be much more confident in our expectation value.  If we have very few
+observations then we are much less confident it's the correct value.
+
+The beta distribution can be used to express the variance in our
+expected leaver percentage.  The parameters for the distribution
+$\alpha$, $\beta$ are simply the observed leavers and non-leavers
+respectively.  Drawing from the distribution is equivalent to working
+out the percentage expectation (obviously slightly changed as it's
+purpose is to introduce variance in this value).  For low values of
+observations there will be a wide range in the values of the expected
+leavers and for high values of observations there will be a much
+narrower range of expected values.
+
+If we use the results of the Beta distribution as the probability
+parameter of the Binomial distribution (composition) we form what's
+called a hierarchical model.  This chaining of distributions increases
+the size of variance in our model runs.
+
+Thus we can write:
+
+\begin{equation*}
+l_{s_x,n_y,ay_z} = \text{Binomial}(t_{s_x,n_y,ay_z},
+\text{Beta}(\alpha^l_{s_x,n_y,ay_z}, \beta^l_{s_x,n_y,ay_z}))
+\end{equation*}
+
+where $\alpha^l_{s_x,n_y,ay_z}$ represents the count of leavers over
+all historic calendar years for the entity $({s_x,n_y,ay_z})$.
+Similarly $\beta^l_{s_x,n_y,ay_z}$ represents the count of all non
+leavers for the same entity over all historic calendar years.
+$(t_{s_x,n_y,ay_z}$ represents the populatio of the entity.
+
+The algorithm described here is a high level overview of the
+implementation by the Send code base.
+
+\subsubsection{Joiners}
+
+The calculation for working out an entity's joiners is split into two
+parts: first the expected rate of joiners is calculated per academic
+year, then this additional population is assigned a need and setting.
+
+The sum of the joiners for a single academic year over all historic
+calendar years is first calculated.
+
+We work out the number of non-joiners by taking the joiners away from
+the predicted number of pupils for an AY (e.g from ONS data).  This
+additional population data source (in addition to the entity state
+that is) is typically only available at an AY level.  Our desire to
+use it means that we can't work at the normal ${s_x,n_y,ay_z}$
+level.
+
+Like for the leavers we wish to have a variance in this value based on
+our confidence of the data.  Like before, we can apply the beta
+distribution to do this.  However, in our code base we divide through
+the sum of the joiners and non-joiners by observed calendar years.
+This has the effect of increasing variance.  It is in direct contrast
+to the other probability distribution calculations that assume iid and
+simply sum these values.
```

Comment:
Note on why we do this? or if we're not sure, note that this is odd and something we expect to investigate further.